From e92bbaed760686cb1e4aa56f7fa1f2ff7ff18bcb Mon Sep 17 00:00:00 2001
From: Sasi Kumar Maddineni <quic_sasikuma@quicinc.com>
Date: Fri, 6 Jun 2025 09:20:39 +0530
Subject: [PATCH 2/2] mount: Allow building when macro LOOP_CONFIGURE is not
 available

This is to allow building the software on machines not having the macro
LOOP_CONFIGURE or the struct loop_config in header "linux/loop.h" (both
of which added to the kernel uapi at the same time); the code snippet
providing them was taken from package util-linux, source file
"include/loopdev.h".

Upstream-Status: Accepted
[https://github.com/containers/composefs/pull/253]

Signed-off-by: Rogerio Guerra Borin <rogerio.borin@toradex.com>
Git-Commit: 384f3068a878ef3c4e29d05a05da3899e111d709
Git-repo: https://github.com/composefs/composefs.git

[Sasi Kumar Maddineni <quic_sasikuma@quicinc.com>: Resolved merge conflicts]
Signed-off-by: Sasi Kumar Maddineni <quic_sasikuma@quicinc.com>
---
 composefs/libcomposefs/lcfs-mount.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/composefs/libcomposefs/lcfs-mount.c b/composefs/libcomposefs/lcfs-mount.c
index 4d1a729..c0ce54c 100644
--- a/composefs/libcomposefs/lcfs-mount.c
+++ b/composefs/libcomposefs/lcfs-mount.c
@@ -49,6 +49,20 @@
 #include "lcfs-utils.h"
 #include "lcfs-internal.h"
 
+#ifndef LOOP_CONFIGURE
+/* Snippet from util-linux/include/loopdev.h
+ *
+ * Since Linux v5.8-rc1 (commit 3448914e8cc550ba792d4ccc74471d1ca4293aae)
+ */
+#define LOOP_CONFIGURE 0x4C0A
+struct loop_config {
+       uint32_t fd;
+       uint32_t block_size;
+       struct loop_info64 info;
+       uint64_t __reserved[8];
+};
+#endif
+
 static int syscall_fsopen(const char *fs_name, unsigned int flags)
 {
 #if defined __NR_fsopen
-- 
2.34.1

